### PR TITLE
Improve k3s navigation / move content around

### PR DESCRIPTION
**Pull request by davidnuzik**
Improve k3s navigation by moving some content around to areas where it makes more sense.
This PR moves content around and only lightly touches / edits files if necessary.

- `Move airgap from advanced to install`
(and also) `Move airgap part 2 (actually del airgap from advanced)`
This moves an air-gap config section we had in the "Advanced Options" section to the "Install Options" section which makes more sense (similar to Rancher docs). This also tweaks the "Install Options" page so that it now lists this and has a hyperlink to it.

- `Move ha backend options from configuration to install`
This moves some content that was in the "Configuration Info" section to the "Install Options" section (HA) which makes more sense. This moves the content to the HA install page at the bottom with this supplemental info that could be important for an HA install.

- `Move Uninstallation notes from advanced to install`
This moves some small content that was in the "Advanced Options" section to the "Install Options" section. This was placed on the main "Install Options" page at the bottom.

- `Fix and move upgrades content to main Install Options section`
In the first commit, where airgap was moved from advanced to install, an "Upgrades" section was accidentally moved as well to airgap. This is fixed and instead moved to the main Install Options section with some light tweaks to content to fix up minor things.

- `Move networking info from Configuration Info to new Networking section`
This moves some content that was in the "Configuration Info" section to a new "Networking" section.

- `Move networking port info from configuration to networking`
This moves some security/network port info from the "Configuration Info" section to the new "Networking" section. Note this content is also already intentionally duplicated on the node requirements page as it has important info about ports that need to be open.

- `REMOVE "Building from Source" section completely`
This completely removes the Building from Source section. After some discussion a while back we agreed it makes sense for this to instead go in the GH README.md so a separate PR will add this in there. Here is the PR to add this into the README.md instead: https://github.com/rancher/k3s/pull/1041

Closes https://github.com/rancher/k3s/issues/859

### CODE REVIEW

**Comment by catherineluse** [comment]
Looks good to me

File: content/k3s/latest/en/installation/ha/_index.md [hunk 1]
```diff
@@ -57,3 +57,99 @@ curl -sfL https://get.k3s.io | K3S_URL=https:/<master_node>:6443 K3S_TOKEN=XXX s
 
 Provide the IP or DNS in place of `<master_node>` this can be any one master node. k3s automatically handles load balancing the master nodes.
 
+# Backend Options
```
Comment (davidnuzik) [inline content/k3s/latest/en/installation/ha/_index.md]: Based on discussion with Catherine, we agreed we should change the heading here to something like "Cluster Datastore Options" to remove ambiguity here.

File: content/k3s/latest/en/installation/ha/_index.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ Installation Outline
 3. Join worker nodes
 
 ### Create Database
```
Comment (davidnuzik) [inline content/k3s/latest/en/installation/ha/_index.md]: Based on discussion with Catherine, we agreed we should change the h3 heading here to something like "Create Database for Cluster Datastore" to remove ambiguity here.

File: content/k3s/latest/en/installation/ha/_index.md (hunk 1 above)
Comment (davidnuzik) [reply content/k3s/latest/en/installation/ha/_index.md]: Resolved in incoming commit.

File: content/k3s/latest/en/installation/ha/_index.md (hunk 2 above)
Comment (davidnuzik) [reply content/k3s/latest/en/installation/ha/_index.md]: Resolved in incoming commit.

**Comment by davidnuzik** [comment]
Resolved comments and has been reviewed and approved. Let's get this merged! :)

**Pull request merged**